pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/IndexLoadingConfig.java

```diff
@@ -870,12 +875,6 @@ public Schema getSchema() {
     return _schema;
   }
 
-  @VisibleForTesting
-  public void setTableConfig(TableConfig tableConfig) {
```

Comment:
👍

I believe logic based on `_dirty = true;` would be fine, as there are a few other methods setting this flag anyway.

Reply:
The main problem is that the setters are not reflected back to table config, which is causing inconsistency across configs